plugins/state_history_plugin/state_history_plugin.cpp

```diff
@@ -84,7 +84,23 @@ struct state_history_plugin_impl : std::enable_shared_from_this<state_history_pl
    state_history::trace_converter   trace_converter;
    std::set<std::shared_ptr<session_base>> session_set;
 
-   using acceptor_type = std::variant<std::unique_ptr<tcp::acceptor>, std::unique_ptr<unixs::acceptor>>;
+   struct tcp_acceptor  {
+      using socket_type = tcp::acceptor::protocol_type::socket;
+      tcp_acceptor(boost::asio::io_context& ioc) : acceptor_(ioc), socket_(ioc), error_timer_(ioc) {}
+      tcp::acceptor               acceptor_;
+      socket_type                 socket_;
+      boost::asio::deadline_timer error_timer_;
+   };
+   
+   struct unix_acceptor {
+      using socket_type = unixs::acceptor::protocol_type::socket;
+      unix_acceptor(boost::asio::io_context& ioc) : acceptor_(ioc), socket_(ioc), error_timer_(ioc) {}
+      unixs::acceptor             acceptor_;
+      socket_type                 socket_;
+      boost::asio::deadline_timer error_timer_;
+   };
+   
```

Comment:
It may be possible to collapse the above in to a templated class; both `ip::tcp` and `local::stream_protocol` have `acceptor` and `socket` typedefs

Reply:
Excellent Matt, I checked in the change!